sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/PlanGenerator.java

```diff
@@ -191,6 +195,16 @@ private Step generateStep(PodInstance podInstance, List<String> tasksToLaunch) {
         }
     }
 
+    /**
+     * Validates that all tasks in a step are defined in the pod.
+     */
+    private boolean isValidStep(PodSpec podSpec, List<String> tasksToLaunch) {
+        List<String> allTaskNames = podSpec.getTasks().stream()
+                .map(taskSpec -> taskSpec.getName())
+                .collect(Collectors.toList());
```

Comment:
Super nitpick: allTaskNames could be a Set (with Collectors.toSet()), and then the following containsAll would be O(n)